query-algebrizer/src/clauses/or.rs

```diff
@@ -78,3 +78,289 @@ fn _simply_matches_value_place(left: &PatternValuePlace, right: &PatternValuePla
         _ => true,
     }
 }
+
+pub enum DeconstructedOrJoin {
+    KnownSuccess,
+    KnownEmpty(EmptyBecause),
+    Unit(OrWhereClause),
+    UnitPattern(Pattern),
+    Simple(Vec<Pattern>),
+    Complex(OrJoin),
+}
+
+/// Application of `or`. Note that this is recursive!
+impl ConjoiningClauses {
+    fn apply_or_join(&mut self, schema: &Schema, mut or_join: OrJoin) -> Result<()> {
+        // Simple optimization. Empty `or` clauses disappear. Unit `or` clauses
```

Comment:
Ah, this partially answers my earlier questions about empty `Intersection` and `Alternation.`  You're going to enforce normalizations that make some of your choices and representations safe.